control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -126,7 +125,67 @@ func (r *MeshGatewayController) onCreateUpdate(ctx context.Context, req ctrl.Req
 		return fmt.Errorf("unable to create role binding: %w", err)
 	}
 
-	// TODO NET-6393
+	//Create Service
+
+	areServicesEqual := func(a, b *corev1.Service) bool {
+
+		if a == nil || b == nil {
+			return true
+		}
+
+		if !equality.Semantic.DeepEqual(a.Annotations, b.Annotations) {
+			return false
+		}
+		if len(b.Spec.Ports) != len(a.Spec.Ports) {
+			return false
+		}
+
+		for i, port := range a.Spec.Ports {
+			otherPort := b.Spec.Ports[i]
+			if port.Port != otherPort.Port {
+				return false
+			}
+			if port.Protocol != otherPort.Protocol {
+				return false
+			}
+		}
+		return true
+	}
+
+	// mergeService is used to keep annotations and ports from the `from` Service
+	// to the `to` service. This prevents an infinite reconciliation loop when
+	// Kubernetes adds this configuration back in.
+	mergeService := func(from, to *corev1.Service) *corev1.Service {
```

Comment:
minor/style pref: could we define this and the `areServicesEqual` as separate named functions rather than inline/anonymous functions (anonymous might not be the right word here, but splitting them out into named functions would allow them to be more easily tested)

Reply:
done :)